crates/types/src/config/worker.rs

```diff
@@ -359,6 +360,32 @@ impl Default for StorageOptions {
 #[serde(rename_all = "kebab-case")]
 #[builder(default)]
 pub struct SnapshotsOptions {
+    /// # Destination
+    ///
+    /// Where snapshots are moved after they get created. This property support URLs with either
+    /// `s3://` or `file://` protocol scheme. The URL is parsed with
+    ///
+    /// Example: `s3://snapshots-bucket/restate/cluster` will send snapshots to the specified
+    /// bucket, prefixing keys with the URL path.
+    ///
+    /// Default: `None`
+    pub destination: Option<String>,
+
+    /// # Additional options
+    ///
+    /// Extra configuration options that will be passed to `object_store`. Supported keys include:
+    ///
+    /// - `region`
+    /// - `access_key_id`
+    /// - `secret_access_key`
+    /// - `session_token`
+    /// - `endpoint_url`
+    ///
+    /// For the full list of available options, please refer to the [Object Store configuration system reference](https://docs.rs/object_store/latest/object_store/index.html#configuration-system)
+    /// and the specific provider's documentation, e.g. [S3](https://docs.rs/object_store/latest/object_store/aws/enum.AmazonS3ConfigKey.html).
+    #[serde(flatten, skip_serializing_if = "HashMap::is_empty")]
+    pub additional_options: HashMap<String, String>,
```

Comment:
Ah, totally didn't see that because I haven't tested this option at all! I'm not sure how to handle it with serde but I'll find a way to make it work.

Reply:
I clearly did all the precursor work but then forgot to delete the field - sorry about that! This is now removed.